src/content/app/genome-browser/components/browser-track-config/useBrowserTrackConfig.ts

```diff
@@ -0,0 +1,258 @@
+/**
+ * See the NOTICE file distributed with this work for additional information
+ * regarding copyright ownership.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { useEffect, useRef } from 'react';
+import { RootState } from 'src/store';
+import { useSelector, useDispatch } from 'react-redux';
+
+import { OptionValue } from 'src/shared/components/radio-group/RadioGroup';
+
+import { getBrowserActiveGenomeId } from '../../state/browser-general/browserGeneralSelectors';
+import {
+  getTrackConfigForTrackId,
+  getApplyToAllConfig,
+  getBrowserCogList,
+  getBrowserSelectedCog
+} from '../../state/track-config/trackConfigSelectors';
+import {
+  updateTrackName as updateTrackConfigTrackName,
+  updateFeatureLabel as updateTrackConfigFeatureLabel,
+  updateShowSeveralTranscripts as updateTrackConfigShowSeveralTranscripts,
+  updateShowTranscriptIds as updateTrackConfigShowTranscriptIds,
+  updateApplyToAll,
+  TrackType
+} from '../../state/track-config/trackConfigSlice';
+
+import analyticsTracking from 'src/services/analytics-service';
+import useGenomeBrowser from '../../hooks/useGenomeBrowser';
+
+const useBrowserTrackConfig = () => {
+  const applyToAllConfig = useSelector(getApplyToAllConfig);
+  const browserCogList = useSelector(getBrowserCogList);
+  const selectedCog = useSelector(getBrowserSelectedCog) || '';
```

Comment:
Think we should not give a default value as empty string here and instead check in `selectedTrackConfigInfo` whether `selectedCog` is `null` before calling `getTrackConfigForTrackId`.